app/src/sharedTest/java/org/oppia/android/app/options/AudioLanguageFragmentTest.kt

```diff
@@ -455,13 +514,29 @@ class AudioLanguageFragmentTest {
     }
   }
 
+  @Test
+  @RunOn(TestPlatform.ROBOLECTRIC)
+  fun testFragment_fromOptions_languageSelectionChanged_selectionIsUpdated() {
+    launchV2FlowWithLanguage(ENGLISH_AUDIO_LANGUAGE, OPTIONS_SCREEN).use { scenario ->
+      scenario.onActivity { activity ->
+        onView(withId(R.id.audio_language_dropdown_list)).perform(click())
+
+        onData(allOf(`is`(instanceOf(String::class.java)), `is`("Naijá")))
+          .inRoot(withDecorView(not(`is`(activity.window.decorView))))
+          .perform(click())
+
+        testCoroutineDispatchers.runCurrent()
+
+        onView(withId(R.id.audio_language_dropdown_list)).check(
+          matches(withText(R.string.nigerian_pidgin_localized_language_name))
+        )
```

Comment:
Does this actually validate the default and not any of the drop-down items? It's a bit hard for me to check just by reading the assertion, so thought I'd double check.

Reply:
No, this is meant to validate that updating the language works. I have added 2 tests that verify defaults.